src/v/config/property.h

```diff
@@ -380,16 +367,81 @@ class binding {
      * the callback and the binding attributes of the same object.
      */
     void watch(std::function<void()>&& f) {
-        oncore_debug_verify(_verify_shard);
+        oncore_debug_verify(binding_base<T>::_verify_shard);
         _on_change = std::move(f);
     }
 
-    const T& operator()() const {
+    // private interface for the property<T> friend
+private:
+    friend class property<T>;
+    void detach() { _parent = nullptr; }
+    void update(const T& v) {
         oncore_debug_verify(_verify_shard);
+        const bool changed = do_update(v);
+        if (changed && _on_change.has_value()) {
+            _on_change.value()();
```

Comment:
a little confused about why this invoked. this would be the conversion the function, right? if so, this would be the conversion happening, but then the result being ignored?

Reply:
> this would be the conversion the function, right?

If you mean `_on_change` then no, it's a user installed watch callback. The conversion has happened in `do_update()`.